contributors/design-proposals/daemonset-create-first-rolling-update.md

```diff
@@ -0,0 +1,182 @@
+
+	// Replace the old daemons only when it's killed
+	OnDeleteDaemonSetStrategyType DaemonSetUpdateStrategyType = "OnDelete"
+)
+
+...
+
+// Spec to control the desired behavior of a daemon set surging rolling update.
+type SurgingRollingUpdateDaemonSet struct {
+	// The maximum number of DaemonSet pods that can be scheduled above the desired number of pods
+	// during the update. Value can be an absolute number (ex: 5) or a percentage of the total number
+	// of DaemonSet pods at the start of the update (ex: 10%). The absolute number is calculated from
+	// the percentage by rounding up. This cannot be 0. The default value is 1. Example: when this is
+	// set to 30%, at most 30% of the total number of nodes that should be running the daemon pod
+	// (i.e. status.desiredNumberScheduled) can have 2 pods running at any given time. The update
+	// starts by starting replacements for at most 30% of those DaemonSet pods. Once the new pods are
+	// available it then stops the existing pods before proceeding onto other DaemonSet pods, thus
+	// ensuring that at most 130% of the desired final number of DaemonSet  pods are running at all
+	// times during the update.
+	// +optional
+	MaxSurge *intstr.IntOrString `json:"maxSurge,omitempty" protobuf:"bytes,2,opt,name=maxSurge"`
+}
+```
+
+Then an additional hook for this strategy will be added to the part of the
+reconciliation loop that runs when pod updates are needed. That hook will:
+
+1. Determine how many nodes can have new pods scheduled, according to the
+   number of extra nodes currently running more than one pod and the `maxSurge`
+   parameter.
+2. Determine how many nodes can have their previous-generation pods retired
+   once the replacement pods are scheduled and running.
+3. Instruct the controller-manager to add and delete pods on nodes based on the
+   results of (1) and (2).
+
+This is mechanically almost identical to the current `RollingUpdate` strategy,
+with the key differences being the direct creation of new pods on nodes (the
+`RollingUpdate` strategy only deletes pods up to the `maxUnavailable` threshold
+and then waits for the main reconciler create new pods to replace them).
+
+One minor complication is that the controller-manager's reconciliation loop
+actively tries to enforce the one-pod-per-node invariant, and work is underway
+to fix cases in which the invariant is not held
+[kubernetes/kubernetes#50477](https://github.com/kubernetes/kubernetes/issues/50477).
+However, this invariant will need to be relaxed when the `SurgingRollingUpdate`
+strategy temporarily creates extra pods.
+
+### Alternatives considered
+
+The `maxSurge` parameter could have been added to the existing `RollingUpdate`
+strategy (as is the case for Deployments). However, this would break validation
+backwards compatibility, since the `maxUnavailable` parameter currently
+requires `maxUnavailable` to be > 0, but this is not required with
+`SurgingRollingUpdate`.
```

Comment:
the amount of confusion a new type of RollingUpdate will create and all the documentation required to explain minor and subtle differences between these new rolling updates will add to users confusion. Was there a way to ask the community what they thought, breaking the validation seems like a not big issue as long as we keep the default behavior backward compatible